src/libraries/shims/fakes/Directory.Build.targets

```diff
@@ -0,0 +1,30 @@
+<Project>
+
+  <Import Project="..\Directory.Build.targets" />
+
+  <PropertyGroup>
+    <TypeForwardedTypesCompilePath>$(IntermediateOutputPath)$(MSBuildProjectName).TypeForwardedTypes.cs</TypeForwardedTypesCompilePath>
+  </PropertyGroup>
+
+  <Target Name="CreateCompileSourceForTypeForwardedTypes"
+          BeforeTargets="CoreCompile"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(TypeForwardedTypesCompilePath)">
+    <Error Text="The 'TypeForwardedType' item must not be empty." Condition="'@(TypeForwardedType)' == ''" />
+
+    <ItemGroup>
+      <TypeForwardedType Namespace="$([System.String]::new('%(Identity)').Substring(0, $([System.String]::new('%(Identity)').LastIndexOf('.'))))" />
+      <TypeForwardedType TypeName="$([System.String]::new('%(Identity)').Substring($([MSBuild]::Add($([System.String]::new('%(Identity)').LastIndexOf('.')), 1))))" />
+      <TypeForwardedTypesLine Include="namespace %(TypeForwardedType.Namespace) { public class %(TypeForwardedType.TypeName) { } }" />
+    </ItemGroup>
+
+    <WriteLinesToFile File="$(TypeForwardedTypesCompilePath)"
+                      Lines="@(TypeForwardedTypesLine)"
+                      Overwrite="true" />
+    
+    <ItemGroup>
+      <Compile Include="$(TypeForwardedTypesCompilePath)" />
+      <FileWrites Include="$(TypeForwardedTypesCompilePath)" />
+    </ItemGroup>
+  </Target>
+</Project> 
```

Comment:
I wonder what we can do to make sure these remain compatible with the OOBs they represent -- or that the shims we produce remain compatible.

Reply:
The shims that we produce are protected by APICompat. If we would accidentally remove a type forward, APICompat should flag that. Are you thinking about additional protection?

Reply:
That's good to know - I wasn't sure if we had that protection, but it's good to know we do.  I wasn't sure we'd catch dropping a typeforward if the target of that forward was missing.

Reply:
Please see my last comment in the PR. The shims were actually not protected by APICompat as assembly references weren't provided by the left side which silences assembly load errors and unresolved type forwards.

My latest commit adds protection for that by feeding the missing assemblies for the left and right comparison into APICompat. It then found one wrong assembly identity (public key token) which I accidentally overlooked in the System.Security diff (https://www.diffchecker.com/7fS43tpW/).

Having that protection is a pretty significant improvement in comparison to what we have in main and it doesn't bring standing prebuilds back as the entire apicompat.proj logic is already excluded by sourcebuild.

Reply:
Thank you for taking a closer look at this.